src/transport/SecureMessageCodec.h

```diff
@@ -47,10 +47,11 @@ namespace SecureMessageCodec {
  * @param msgBuf        The message buffer that contains the unencrypted message. If
  *                      the operation is successful, this buffer will be mutated to contain
  *                      the encrypted message.
+ * @param inputMaxLength Max size for input
```

Comment:
What is the point of this argument?  Why wouldn't callers just pass `msgBuf->TotalLength()` for it?

Either this argument should not exist, or the documentation needs to do a _much_ better job of explaining what this is for and how it should be used.

My temptation is towards the former: why does the crypto stuff care about this length?  It doesn't.

Reply:
so this argument is meant to provide the method the maximum length of payload's length
that means we can't just pass msgBuf->TotalLength() because that would bypass the check :) , i think this place is okay since that check is common for all the places that's calling SecureMessageCodec::Encrypt(